logging/logger.go

```diff
@@ -42,15 +42,43 @@ var (
 	defaultLoggerOnce sync.Once
 )
 
+// Factory is a factory to create loggers.
+type Factory struct {
```

Comment:
To use the existing logging package with the CLI flags, one would need to use this factory instead.

```go
lf := &logging.Factory{}
fs.LoggerVar(lf)
logger := lf.New()
```

Reply:
I'm not sure I follow why we need this?

Reply:
`zap.SugaredLogger` doesn't provide a way to set level or mode dynamically (see my thinking in the PR description). A factory is the next best thing I could thought of

Reply:
Ah I see. What about when we move to slog? We might want to wait the ~month until 1.21 is released and just do this once with the slog package instead of trying to do something more general.

Reply:
sg! Let me comment in the slog PR so don't forget.